packages/payload/src/fields/validations.ts

```diff
@@ -269,6 +270,10 @@ const validateFilterOptions: Validate = async (
               })
             : filterOptions
 
+        if (optionFilter === true) {
+          optionFilter = {}
```

Comment:
What about doing `optionFilter = null` ?

That way, the unnecessary `findWhere.and.push(optionFilter)` would be skipped here